modules/weko-index-tree/weko_index_tree/api.py

```diff
@@ -862,6 +863,31 @@ def recursive_p():
         q = query.order_by(recursive_t.c.path).all()
         return q
 
+    @classmethod
+    def get_child_list_by_index(cls, pid):
+        """
+        Get index list info.
+
+        :param pid: pid of the index.
+        :return: the list of index.
+        """
+        recursive_t = db.session.query(
+            Index.parent.label("pid"),
+            Index.id.label("cid"),
+        ).filter(Index.id == pid). \
+            cte(name="recursive_t", recursive=True)
+
+        rec_alias = aliased(recursive_t, name="rec")
+        test_alias = aliased(Index, name="t")
+        recursive_t = recursive_t.union_all(
+            db.session.query(
+                test_alias.parent,
+                test_alias.id,
+            ).filter(test_alias.parent == rec_alias.c.cid)
+        )
+        query = db.session.query(recursive_t).all()
+        return query
+
```

Comment:
Existing function `get_child_list_by_pip `and this function have same parameter, expectation. Why don't you use existing function and then get the value you need instead of writing a new function?